app/src/main/java/io/intrepid/contest/screens/contestcreation/addcategoriestocontest/AddCategoriesPresenter.java

```diff
@@ -0,0 +1,48 @@
+package io.intrepid.contest.screens.contestcreation.addcategoriestocontest;
+
+import android.support.annotation.NonNull;
+
+import java.util.List;
+
+import io.intrepid.contest.base.BasePresenter;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.models.Category;
+import io.intrepid.contest.models.Contest;
+import io.intrepid.contest.models.ContestProvider;
+
+import static io.intrepid.contest.screens.contestcreation.addcategoriestocontest.AddCategoriesContract.View;
+
+
+class AddCategoriesPresenter extends BasePresenter<AddCategoriesContract.View> implements AddCategoriesContract.Presenter {
+    private List<Category> categories;
+    private Category editableCategory;
+
+    AddCategoriesPresenter(@NonNull View view, @NonNull PresenterConfiguration configuration) {
+        super(view, configuration);
+        categories = ContestProvider.getInstance().getLastEditedContest().categories;
+        editableCategory = new Category();
```

Comment:
I am not sure if this applies to singleton classes like `ContestProvider` (which could be mocked), and in this case that the category will always be a new instance... But wouldn't it be easier to test if `categories` and `editableCategory` were passed to the constructor by the view? @jmccormack200